packages/core/src/components/toast/toaster.tsx

```diff
@@ -133,6 +142,12 @@ export class Toaster extends AbstractPureComponent2<IToasterProps, IToasterState
     private toastId = 0;
 
     public show(props: IToastProps, key?: string) {
+        // show no toasts if max toasts is less than 1.
+        if (this.props.maxToasts < 1) {
```

Comment:
this validation should happen in `validateProps` (invoked via the super class `AbstractPureComponent2`). if `props.maxToasts < 1` I think you should fail loudly and throw a validation error (some other components do this).